### PR TITLE
WIP: add auto zignspace generation for the zg command

### PR DESCRIPTION
**Pull request by AlexKornitzer**
Part 1:
This allows the 'zg' command to auto generate zignspaces from function
names by utilising the ':' convention used in zignatures. Now when 'zg' is
run if any funtion names are in the format ZIGNSPACE:FUNCTION, the
zignspace will be extracted, created and the function applied under its
scope.

### CODE REVIEW

**Comment by AlexKornitzer** [comment]
This starts the work on #13934

**Comment by AlexKornitzer** [comment]
Due to the current implementation we kind of get the zignspace in the match for free due to it being in the name. Are there any strong views on this, should the ':' be changed to '.' or are we happy with it as is?

image

Furthermore should we strip the zignspace from the function name when printing? Should be easy to do as we can just prefix check, otherwise we currently have the following output:
image

File: libr/core/cmd_zign.c [hunk 1]
```diff
@@ -183,13 +183,26 @@ static char *getFcnComments(RCore *core, RAnalFunction *fcn) {
...
+		// If the user has set funtion names containing a single ':' then we assume
+		// ZIGNSPACE:FUNCTION, and for now we only support the 'zg' command
+		if (r_str_char_count (fcn->name, ':') == 1) {
+			for (i = 0; i < strlen (fcn->name); i++) {
+				if (fcn->name[i] == ':') {
+					break;
+				}
+				len++;
+			}
+			zignspace = r_str_newlen(fcn->name, len);
```
Comment (radare) [inline libr/core/cmd_zign.c]: missing space before (

File: libr/core/cmd_zign.c [hunk 2]
```diff
@@ -183,13 +183,26 @@ static char *getFcnComments(RCore *core, RAnalFunction *fcn) {
...
+	int i, len = 0;
 
 	if (name) {
 		zigname = r_str_new (name);
 	} else {
-		if (curspace) {
+		// If the user has set funtion names containing a single ':' then we assume
+		// ZIGNSPACE:FUNCTION, and for now we only support the 'zg' command
+		if (r_str_char_count (fcn->name, ':') == 1) {
+			for (i = 0; i < strlen (fcn->name); i++) {
```
Comment (radare) [inline libr/core/cmd_zign.c]: put the strlen outside the loop

Comment (radare) [inline libr/core/cmd_zign.c]: uhm. what about just using strchr ?

File: libr/core/cmd_zign.c (hunk 1 above)
Comment (AlexKornitzer) [reply libr/core/cmd_zign.c]: Done

File: libr/core/cmd_zign.c (hunk 2 above)
Comment (AlexKornitzer) [reply libr/core/cmd_zign.c]: Removed

Comment (AlexKornitzer) [reply libr/core/cmd_zign.c]: Yeah makes much more sense my C is incredibly rust atm, worth checking I have not messed up the arithmetic there.

**Comment by AlexKornitzer** [comment]
So all good to go, unless we address my queries above about:
1. Using ':' in flags
2. Removing zignspace from zignature name

**Comment by radare** [comment]
sgtm, lets wait for travis to be green , did u rebased?

**Comment by AlexKornitzer** [comment]
Ah yes will rebase and squash. I think I will do a new PR for addressing the zignzpace in zignature name issue as IMO is should be removed.

**Pull request merged**